app/channel.go

```diff
@@ -2337,7 +2317,17 @@ func (a *App) removeUserFromChannel(c *request.Context, userIDToRemove string, r
 				return model.NewAppError("removeUserFromChannel", "api.team.remove_user_from_team.missing.app_error", nil, err.Error(), http.StatusBadRequest)
 			}
 
-			if err = a.RemoveTeamMemberFromTeam(c, teamMember, removerUserId); err != nil {
+			// Send the websocket message before we actually do the remove so the user being removed gets it.
+			message := model.NewWebSocketEvent(model.WebsocketEventLeaveTeam, teamMember.TeamId, "", "", nil)
+			message.Add("user_id", teamMember.UserId)
+			message.Add("team_id", teamMember.TeamId)
+			a.Publish(message)
+
+			if err := a.srv.teamService.RemoveTeamMember(teamMember); err != nil {
+				return model.NewAppError("removeUserFromChannel", "api.team.remove_user_from_team.missing.app_error", nil, err.Error(), http.StatusBadRequest)
+			}
+
+			if err = a.postProcessTeamMemberLeave(c, teamMember, removerUserId); err != nil {
```

Comment:
Looks like this logic is duplicated in app/channel.go and app/team.go. Can we refactor this to a common method?

Reply:
Sure 👍